File: commands.py
import webbrowser
import urllib.parse
from datetime import datetime


def open_url(url):
    try:
        webbrowser.open(url)
        return True
    except Exception:
        return False


def open_youtube():
    return open_url("https://www.youtube.com")


def open_google():
    return open_url("https://www.google.com")


def open_gmail():
    return open_url("https://mail.google.com")


def open_whatsapp():
    return open_url("https://web.whatsapp.com")


def google_search(query):
    if not query:
        return False

    query = urllib.parse.quote_plus(query)
    return open_url(
        f"https://www.google.com/search?q={query}"
    )


def youtube_search(query):
    if not query:
        return False

    query = urllib.parse.quote_plus(query)
    return open_url(
        f"https://www.youtube.com/results?search_query={query}"
    )


def gmail_search(query):
    if not query:
        return open_gmail()

    query = urllib.parse.quote_plus(query)
    return open_url(
        f"https://mail.google.com/mail/u/0/#search/{query}"
    )


def get_time():
    return datetime.now().strftime("%I:%M %p")


def get_date():
    return datetime.now().strftime("%d %B %Y")


def call_contact(contact):
    return {
        "action": "call",
        "contact": contact,
        "status": "phone_bridge_pending"
    }


def send_whatsapp(contact, message):
    return {
        "action": "whatsapp",
        "contact": contact,
        "message": message,
        "status": "phone_bridge_pending"
    }


def execute_command(command, argument=""):

    command = command.lower().strip()
    argument = argument.strip()

    if command == "open_youtube":
        return open_youtube()

    if command == "open_google":
        return open_google()

    if command == "open_gmail":
        return open_gmail()

    if command == "open_whatsapp":
        return open_whatsapp()

    if command == "google_search":
        return google_search(argument)

    if command == "youtube_search":
        return youtube_search(argument)

    if command == "gmail_search":
        return gmail_search(argument)

    if command == "get_time":
        return get_time()

    if command == "get_date":
        return get_date()

    if command == "call":
        return call_contact(argument)

    return False

File: commands.js
// ============================================================
// JARVIS 3.2
// commands.js
// Command & Action Engine
// ============================================================

async function executeCommand(command, argument) {

    switch (command) {

        // =========================
        // YOUTUBE
        // =========================

        case "open_youtube":
            window.open("https://www.youtube.com", "_blank");
            break;

        case "youtube_search":
            if (argument) {
                const youtubeURL =
                    "https://www.youtube.com/results?search_query=" +
                    encodeURIComponent(argument);

                window.open(youtubeURL, "_blank");
            }
            break;


        // =========================
        // GOOGLE
        // =========================

        case "open_google":
            window.open("https://www.google.com", "_blank");
            break;

        case "google_search":
            if (argument) {
                const googleURL =
                    "https://www.google.com/search?q=" +
                    encodeURIComponent(argument);

                window.open(googleURL, "_blank");
            }
            break;


        // =========================
        // GMAIL
        // =========================

        case "open_gmail":
            window.open("https://mail.google.com", "_blank");
            break;

        case "gmail_search":
            if (argument) {
                const gmailURL =
                    "https://mail.google.com/mail/u/0/#search/" +
                    encodeURIComponent(argument);

                window.open(gmailURL, "_blank");
            }
            break;


        // =========================
        // WHATSAPP
        // =========================

        case "open_whatsapp":
            window.open("https://web.whatsapp.com", "_blank");
            break;


        // =========================
        // TIME
        // =========================

        case "get_time":
            console.log(
                "Current time:",
                new Date().toLocaleTimeString()
            );
            break;


        // =========================
        // DATE
        // =========================

        case "get_date":
            console.log(
                "Current date:",
                new Date().toLocaleDateString()
            );
            break;


        // =========================
        // CALL
        // =========================

        case "call":

            if (argument) {

                const phoneNumber =
                    argument.replace(/[^0-9+]/g, "");

                if (phoneNumber) {
                    window.location.href =
                        "tel:" + phoneNumber;
                } else {
                    console.log(
                        "A phone number is required for calling."
                    );
                }

            }

            break;


        // =========================
        // UNKNOWN COMMAND
        // =========================

        default:

            console.log(
                "Unknown Jarvis command:",
                command,
                argument
            );
    }
}


// Make the function available globally
window.executeCommand = executeCommand;
